komposable-architecture/src/main/java/com/toggl/komposable/architecture/Effect.kt

```diff
@@ -14,5 +21,25 @@ fun interface Effect<out Action> {
      * @return An action that will be sent again for further processing
      * @see Store.send
      */
-    suspend fun execute(): Action?
+    operator fun invoke(): Flow<Action>
```

Comment:
I don't really like this change. While in the context of my pet projects doing operator overloading is fine, once you have too much of it you create needless layers of indirection. I don't think calling `effect.execute()` is so bad that you wanna omit the `.execute`, specially since this won't be something that the end users need to do often (likely only in tests, where readability is usually important). 

That said, this is not a hill I'd die on 🤷

Reply:
@heytherewill  Yeah, I don't think the extra syntactic sugar that `invoke()` provides is important; I just didn't know how else to call this tbh :-D 
I just wasn't sure if "execute" correctly describes a function that returns flow where it can be technically "executed" later in some sense 🤷 
Can you think of a better name?

Reply:
`run`? Not sure if because it already exists in another context it would be confusing. `start` would also be a good call

Reply:
I guess both could work, I like `run` better probably, but as you said, `run` from stdlib makes it a bit confusing. What about just `actions` or `actionFlow`. Idk, naming is so hard, help us @daividssilverio, @thuutin

Reply:
I like `start` and `actions`, but could be `flow`, `process`, `results`, `produce`, `enqueue`, `breakTheWorld`

`start` and `process` being my favorites. Just `actions` sounds less than it actually is as we can do IO and other stuff inside effects.